blocks/api/templates.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import { every, map } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { createBlock } from './factory';
+
+/**
+ * Checks whether a list of blocks matches a template by comparing the block names.
+ *
+ * @param {Array} blocks    Block list.
+ * @param {Array} template  Block template.
+ *
+ * @return {boolean}        Whether the list of blocks matches a templates
+ */
+export function doesBlocksMatchTemplate( blocks = [], template = [] ) {
+	return blocks.length === template.length && every( template, ( [ name,, innerBlocksTemplate ], index ) => {
```

Comment:
Aversion to long lines makes me wish this was:

```js
if ( blocks.length !== template.length ) {
	return false;
}

return every( /* ... */ );
```

Reply:
These double commas can be easy to overlook. Personally I add a blank space between them, though not really sure it's any better. Thoughts?